src/sentry/static/sentry/app/views/releases/detail/releaseActions.tsx

```diff
@@ -1,65 +1,144 @@
 import React from 'react';
 import styled from '@emotion/styled';
-import {browserHistory} from 'react-router';
 
-import {t} from 'app/locale';
+import {t, tct, tn} from 'app/locale';
+import {Release} from 'app/types';
 import space from 'app/styles/space';
 import Button from 'app/components/button';
-import {IconDelete} from 'app/icons';
+import {IconEllipsis} from 'app/icons';
 import Confirm from 'app/components/confirm';
-import {
-  addErrorMessage,
-  addLoadingMessage,
-  addSuccessMessage,
-} from 'app/actionCreators/indicator';
+import DropdownLink from 'app/components/dropdownLink';
+import MenuItem from 'app/components/menuItem';
+import ProjectBadge from 'app/components/idBadge/projectBadge';
+import {formatVersion} from 'app/utils/formatters';
 import Tooltip from 'app/components/tooltip';
+import TextOverflow from 'app/components/textOverflow';
 
-import {deleteRelease} from './utils';
+import {archiveRelease, restoreRelease} from './utils';
+import {isReleaseArchived} from '../utils';
 
 type Props = {
   orgId: string;
-  version: string;
-  hasHealthData: boolean;
+  release: Release;
+  refetchData: () => void;
 };
 
-const ReleaseActions = ({orgId, version, hasHealthData}: Props) => {
-  const handleDelete = async () => {
-    const redirectPath = `/organizations/${orgId}/releases/`;
-    addLoadingMessage(t('Deleting Release...'));
-
-    try {
-      await deleteRelease(orgId, version);
-      addSuccessMessage(t('Release was successfully removed.'));
-      browserHistory.push(redirectPath);
-    } catch (error) {
-      const errorMessage =
-        error.responseJSON?.detail ?? t('Release could not be be removed.');
-      addErrorMessage(errorMessage);
-    }
-  };
+function ReleaseActions({orgId, release, refetchData}: Props) {
+  function handleArchive() {
+    archiveRelease(orgId, release.version);
+  }
+
+  function handleRestore() {
+    restoreRelease(orgId, release.version, refetchData);
+  }
+
+  function getProjectList() {
+    const maxVisibleProjects = 5;
+    const visibleProjects = release.projects.slice(0, maxVisibleProjects);
+    const numberOfCollapsedProjects = release.projects.length - visibleProjects.length;
+
+    return (
+      <React.Fragment>
+        {visibleProjects.map(project => (
+          <ProjectBadge key={project.slug} project={project} avatarSize={18} />
+        ))}
+        {numberOfCollapsedProjects > 0 && (
+          <span>
+            <Tooltip
+              title={release.projects
+                .slice(maxVisibleProjects)
+                .map(p => p.slug)
+                .join(', ')}
+            >
+              + {tn('%s other project', '%s other projects', numberOfCollapsedProjects)}
+            </Tooltip>
+          </span>
+        )}
+      </React.Fragment>
+    );
+  }
+
+  function getModalHeader(title: React.ReactNode) {
+    return (
+      <h4>
+        <TextOverflow>{title}</TextOverflow>
+      </h4>
+    );
+  }
+
+  function getArchiveModalMessage() {
```

Comment:
should we maybe unit `getArchiveModalMessage` and `getRestoreModalMessage` and only pass a parameter with a different message like we did in the `getModalHeader` function? they look very similar